sdk/program/src/bpf_loader_upgradeable.rs

```diff
@@ -192,6 +192,14 @@ pub fn is_upgrade_instruction(instruction_data: &[u8]) -> bool {
     3 == instruction_data[0]
 }
 
+pub fn is_set_authority_instruction(instruction_data: &[u8]) -> bool {
+    4 == instruction_data[0]
+}
+
+pub fn is_cpi_authorized_instruction(instruction_data: &[u8]) -> bool {
```

Comment:
Get rid of this and move this logic to syscalls.rs, would like to keep the decision logic in one place

Reply:
Done in 454931c